fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/LogicalMetric.java

```diff
@@ -74,4 +74,13 @@
      * @return A collection of metric identity metadata.
      */
     LogicalMetricInfo getLogicalMetricInfo();
+
+    /**
+     * Returns a copy of this logical metric with the provided info replacing the existing info. The original metric
+     * remains untouched.
+     *
+     * @param info  The new metric info for the resulting LogicalMetric
+     * @return a copy of this metric with the new info
+     */
+    LogicalMetric withLogicalMetricInfo(LogicalMetricInfo info);
```

Comment:
I'm not going to block on this or anything, because it's unavoidable, just
want to warn that this *is* a pretty significant change for customers, much
more so than adding a method to an interface would appear to be. Anybody who
extends LogicalMetricImpl is going to need to make sure to override this method
so that they're returning the right type of LogicalMetric. We should make sure
to call out in the CHANGELOG that anybody extending LogicalMetricImpl should
make sure to override this method.

Reply:
on the bright side it's a compile error so it'll hit them fast.  Are you extending LMImpl?  Anyone who's been hanging back hasn't even absorbed the LogicalMetric ->  Interface + Implementation.

Reply:
Maybe we should pop a middle version soon after all this churn

Reply:
@michael-mclawhorn We just recently upgraded to a recent version of Fili, and so absorbed the Interface + Iimplementation change already. So this won't actually be a compiler error for us, since `LogicalMetricImpl` implements the method, and we're extending `LogicalMetricImpl`.

Reply:
Understood.  If your code goes through one of the transforms, such as by picking up a protocol (intentionally or otherwise), your metric's individuality will be erased.

If I may ask, what do you use subclassing to add to your metric?

Reply:
I'm just thinking out loud here, but if the public method called a private method would that force the issue, or would it do the same thing it does here?